csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/DependencyManager.cs

```diff
@@ -783,13 +784,38 @@ private void RestoreProjects(IEnumerable<string> projects, out IEnumerable<strin
             CompilationInfos.Add(("Successfully restored project files", successCount.ToString()));
         }
 
+        [GeneratedRegex(@"^(.+)\.(\d+\.\d+\.\d+(-(.+))?)$", RegexOptions.IgnoreCase | RegexOptions.Compiled | RegexOptions.Singleline)]
+        private static partial Regex LegacyNugetPackage();
+
         private void DownloadMissingPackages(List<FileInfo> allFiles, ISet<string> dllPaths)
         {
             var alreadyDownloadedPackages = Directory.GetDirectories(packageDirectory.DirInfo.FullName)
                 .Select(d => Path.GetFileName(d).ToLowerInvariant());
             var notYetDownloadedPackages = fileContent.AllPackages
                 .Except(alreadyDownloadedPackages)
-                .ToList();
+                .ToHashSet();
+
+            var oldPackageDirectories = Directory.GetDirectories(legacyPackageDirectory.DirInfo.FullName)
+                .Select(d => Path.GetFileName(d).ToLowerInvariant());
+            foreach (var oldPackageDirectory in oldPackageDirectories)
+            {
+                // nuget install restores packages to 'packagename.version' folders (dotnet restore to 'packagename/version' folders)
+                // typical folder names look like:
+                //   newtonsoft.json.13.0.3
+                // there are more complex ones too, such as:
+                //   runtime.tizen.4.0.0-armel.Microsoft.NETCore.DotNetHostResolver.2.0.0-preview2-25407-01
+
+                var match = LegacyNugetPackage().Match(oldPackageDirectory);
+                if (!match.Success)
+                {
+                    logger.LogWarning($"Package directory '{oldPackageDirectory}' doesn't match the expected pattern.");
+                    continue;
+                }
+
+                var packageName = match.Groups[1].Value.ToLowerInvariant();
+                notYetDownloadedPackages.Remove(packageName);
+            }
```

Comment:
Would it be possible to refactor this into a separate method that returns the list of "legacy" packages that matches the legacy nuget format (say `legacyNuget`).
Then 
```csharp
var notYetDownloadedPackages = fileContent.AllPackages
                .Except(alreadyDownloadedPackages)
                .Except(legacyNuget)
                .ToList()
```